sdk/eventhub/azure-eventhub/TROUBLESHOOTING.md

```diff
@@ -0,0 +1,228 @@
+
+This troubleshooting guide covers failure investigation techniques, common errors for the credential types in the Azure Event Hubs Python client library, and mitigation steps to resolve these errors.
+
+- [Handle Event Hubs exceptions](#handle-event-hubs-exceptions)
+  - [Find relevant information in exception messages](#find-relevant-information-in-exception-messages)
+  - [Commonly encountered exceptions](#commonly-encountered-exceptions)
+- [Permission issues](#permission-issues)
+- [Connectivity issues](#connectivity-issues)
+  - [Timeout when connecting to service](#timeout-when-connecting-to-service)
+  - [SSL handshake failures](#ssl-handshake-failures)
+  - [Socket exhaustion errors](#socket-exhaustion-errors)
+  - [Connect using an IoT connection string](#connect-using-an-iot-connection-string)
+  - [Cannot add components to the connection string](#cannot-add-components-to-the-connection-string)
+- [Enable and configure logging](#enable-and-configure-logging)
+  - [Configuring Log4J 2](#configuring-log4j-2)
+  - [Configuring logback](#configuring-logback)
+  - [Enable AMQP transport logging](#enable-amqp-transport-logging)
+  - [Reduce logging](#reduce-logging)
+- [Troubleshoot EventProducerAsyncClient/EventProducerClient issues](#troubleshoot-eventproducerasyncclienteventproducerclient-issues)
+  - [Cannot set multiple partition keys for events in EventDataBatch](#cannot-set-multiple-partition-keys-for-events-in-eventdatabatch)
+  - [Setting partition key on EventData is not set in Kafka consumer](#setting-partition-key-on-eventdata-is-not-set-in-kafka-consumer)
+- [Troubleshoot EventProcessorClient issues](#troubleshoot-eventprocessorclient-issues)
+  - [412 precondition failures when using an event processor](#412-precondition-failures-when-using-an-event-processor)
+  - [Partition ownership changes frequently](#partition-ownership-changes-frequently)
+  - ["...current receiver '<RECEIVER_NAME>' with epoch '0' is getting disconnected"](#current-receiver-receiver_name-with-epoch-0-is-getting-disconnected)
+  - [High CPU usage](#high-cpu-usage)
+  - [Processor client stops receiving](#processor-client-stops-receiving)
+  - [Migrate from legacy to new client library](#migrate-from-legacy-to-new-client-library)
+- [Get additional help](#get-additional-help)
+  - [Filing GitHub issues](#filing-github-issues)
+
+## Handle Event Hubs exceptions
+
+All Event Hubs exceptions are wrapped in an [EventHubError][EventHubError].  They often have an underlying AMQP error code which specifies whether an error should be retried.  For retryable errors (ie. `amqp:connection:forced` or `amqp:link:detach-forced`), the client libraries will attempt to recover from these errors based on the [retry options][AmqpRetryOptions] specified when instantiating the client.  To configure retry options, follow the sample [Client Creation][ClientCreation].  If the error is non-retryable, there is some configuration issue that needs to be resolved.
```

Comment:
Same basic thought as Swathi's below.   Does Python expose these strings as the error message?    If no, I think we can just remove them here without replacements.